pkg/ipcache/ipcache.go

```diff
@@ -598,7 +613,10 @@ func (ipc *IPCache) RemovePrefixes(prefixes []netip.Prefix, src source.Source, r
 	}
 	ipc.metadata.Unlock()
 	ipc.metadata.enqueuePrefixUpdates(affectedPrefixes...)
-	ipc.TriggerLabelInjection()
+	// If IP cache is used without data plane, we should not trigger label injection.
```

Comment:
Data plane subscribes to updates through `IPIdentityMappingListener` interface, so if it doesn't register in the first place we shouldn't need to sprinkle extra conditional checks into every operation that the ipcache does. Basically I'm wondering if there's a cleaner way to achieve the same without putting additional cognitive load on maintainers trying to reason about the ipcache logic.